jerry-core/jerry-snapshot.c

```diff
@@ -630,3 +632,409 @@ jerry_exec_snapshot (const void *snapshot_p, /**< snapshot */
+  ecma_compiled_code_t *bytecode_data_p;
+  parse_status = parser_parse_script (source_p,
+                                      source_size,
+                                      is_strict,
+                                      &bytecode_data_p);
+
+  if (ECMA_IS_VALUE_ERROR (parse_status))
+  {
+    ecma_free_value (parse_status);
+    return 0;
+  }
+
+  ecma_free_value (parse_status);
+  ecma_bytecode_deref (bytecode_data_p);
+
+  ecma_lit_storage_item_t *string_list_p = JERRY_CONTEXT (string_list_first_p);
+  lit_utf8_size_t literal_count = 0;
+  size_t total_bytes = 0;
+
+  /* Count the valid and non-magic identifiers in the list */
+  while (string_list_p != NULL)
+  {
+    for (int i = 0; i < ECMA_LIT_STORAGE_VALUE_COUNT; i++)
+    {
+      if (string_list_p->values[i] != JMEM_CP_NULL)
+      {
+        ecma_string_t *literal_p = JMEM_CP_GET_NON_NULL_POINTER (ecma_string_t,
+                                                                 string_list_p->values[i]);
+        /* We don't save a literal which isn't a valid identifier
+           or it's a magic string */
+        if (ecma_string_is_valid_identifier (literal_p)
+            && ecma_get_string_magic (literal_p) == LIT_MAGIC_STRING__COUNT)
+        {
+          literal_count++;
+        }
+      }
+    }
+
+    string_list_p = JMEM_CP_GET_POINTER (ecma_lit_storage_item_t, string_list_p->next_cp);
+  }
+
+  if (literal_count == 0)
+  {
+    return 0;
+  }
+
+  ecma_string_t *literal_array[literal_count];
+  ecma_string_t *literal_array_sorted[literal_count];
+  lit_utf8_size_t literal_idx = 0;
+
+  string_list_p = JERRY_CONTEXT (string_list_first_p);
+
+  while (string_list_p != NULL)
+  {
+    for (int i = 0; i < ECMA_LIT_STORAGE_VALUE_COUNT; i++)
+    {
+      if (string_list_p->values[i] != JMEM_CP_NULL)
+      {
+        ecma_string_t *literal_p = JMEM_CP_GET_NON_NULL_POINTER (ecma_string_t,
+                                                                 string_list_p->values[i]);
+
+        if (ecma_string_is_valid_identifier (literal_p)
+            && ecma_get_string_magic (literal_p) == LIT_MAGIC_STRING__COUNT)
+        {
+          literal_array[literal_idx++] = literal_p;
+        }
+      }
+    }
+
+    string_list_p = JMEM_CP_GET_POINTER (ecma_lit_storage_item_t, string_list_p->next_cp);
+  }
+
+  /* Sort the strings by size at first, then lexicographically */
+  jerry_save_literals_sort_helper (literal_array, literal_array_sorted, 0, literal_count - 1);
+
+  uint8_t * const buffer_end_p = buffer_p + buffer_size;
+
+  if (is_c_format)
+  {
+    /* Save literal count */
+    buffer_p += append_chars_to_buffer (buffer_p,
+                                        buffer_end_p,
+                                        &total_bytes,
+                                        (const char *) "jerry_length_t literal_count = ",
+                                        0);
+
+    buffer_p += append_number_to_buffer (buffer_p, buffer_end_p, &total_bytes, literal_count);
```

Comment:
When the buffer is insufficient, we cannot determine the exact amount of the used bytes just from the pointers. But if we don't want to save anything in the mentioned case, we can omit this variable.